scipy/optimize/_highs/meson.build

```diff
@@ -1,290 +1,75 @@
-highs_define_macros = [
-  '-DCMAKE_BUILD_TYPE="RELEASE"',
-  '-DFAST_BUILD=ON',
-  '-DHIGHS_GITHASH="n/a"',
-  '-DHIGHS_COMPILATION_DATE="2021-07-09"',  # cannot generate dynamically
-  '-DHIGHS_VERSION_MAJOR=1',  # don't care about this, look at CMakelists.txt
-  '-DHIGHS_VERSION_MINOR=2',
-  '-DHIGHS_VERSION_PATCH=0',
-  '-DHIGHS_DIR=' + meson.current_source_dir() / '..' / '..' / '_lib' / 'highs',
-  '-UOPENMP',
-  '-UEXT_PRESOLVE',
-  '-USCIP_DEV',
-  '-UHiGHSDEV',
-  '-UOSI_FOUND',
-  '-DNDEBUG'
-]
+fs = import('fs')
+if not fs.exists('../../../subprojects/highs/README.md')
+    error(
+        'Missing the `highs` submodule! Run `git submodule update --init` to fix this.',
+    )
+endif
```

Comment:
is this needed? I think Meson should error if this is the case and highs is not a system library (https://mesonbuild.com/Reference-manual_functions.html#subproject)

Reply:
Yes, but I believe this was done to keep the build system consistent, i.e. even people unfamiliar with `meson` should be able to continue to get a working checkout with the old submodule instructions.

I had initially added a `.wrap` file since that's the more `meson`-centric approach (and easier to handle IMO) but there was a request to bring back the submodules (and more context as to why) over in https://github.com/scipy/scipy/pull/19255#issuecomment-1745605754 and subsequent messages.

Reply:
thanks, I suppose that answers my question at https://github.com/mesonbuild/meson/discussions/13683. Looks like it boils down to "it doesn't really matter which you use, but staying with a submodule is probably less disruption for SciPy".

In gh-21270, I need a wrap file anyway to invoke CMake, so I wonder whether having a submodule as well is still preferable to using `[wrap-git]`.

Reply:
we'd have to remove this if we wanted to support using highs as a system library though, right?

Reply:
Yup, or rather make it conditional on another meson option. I don't think we're likely to allow that anytime soon though, highs moves far too quickly, which is why we vendor it under Scipy and also why we don't directly import highspy